src/Crate/DBAL/Platforms/CratePlatform.php

```diff
@@ -186,34 +186,63 @@ public function getListTablesSQL()
             "WHERE schema_name = 'doc' OR schema_name = 'blob'";
     }
 
-    protected function tableAndSchema($fqnRelationName) {
-        $t = explode('.', $fqnRelationName);
-        if (count($t) == 1) {
-            array_unshift($t, 'doc');
-        }
-        return $t;
-    }
-
     /**
      * {@inheritDoc}
      */
     public function getListTableColumnsSQL($table, $database = null)
     {
-        $t = $this->tableAndSchema($table);
-        // todo: make safe
-        return "SELECT * from information_schema.columns " .
-            "WHERE table_name = '$t[1]' AND schema_name = '$t[0]'";
+        return "SELECT * from information_schema.columns c " .
+                "WHERE " . $this->getTableWhereClause($table);
     }
 
     /**
      * {@inheritDoc}
      */
     public function getListTableConstraintsSQL($table, $database = null)
     {
-        $t = $this->tableAndSchema($table);
-        // todo: make safe
-        return "SELECT constraint_name, constraint_type from information_schema.table_constraints " .
-            "WHERE table_name = '$t[1]' AND schema_name = '$t[0]' AND constraint_type = 'PRIMARY KEY'";
+        return "SELECT constraint_name, constraint_type from information_schema.table_constraints c " .
+            "WHERE " . $this->getTableWhereClause($table) . " AND constraint_type = 'PRIMARY KEY'";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getListTableIndexesSQL($table, $currentDatabase = null)
+    {
+        return "SELECT c.constraint_name, c.constraint_type, k. column_name from information_schema.table_constraints c " .
+                "JOIN information_schema.key_column_usage k on c.constraint_name = k.constraint_name " .
+                "WHERE " . $this->getTableWhereClause($table);
+    }
+
+    private function getTableWhereClause($table, $tableAlias = 'c')
+    {
+        if (strpos($table, '.') !== false) {
+            [$schema, $table] = explode('.', $table);
+            $schema = $this->quoteStringLiteral($schema);
+        } else {
+            $schema = $this->quoteStringLiteral('doc');
+        }
+
+        $table = new Identifier($table);
```

Comment:
What is the difference between creating an `Identifier` instance first and using the string value directly?

```php
$table = new Identifier($table);
$table = $this->quoteStringLiteral($table->getName());
```

vs

```php
$table = $this->quoteStringLiteral($table);
```

Asking because the schema is quoted directly.

Reply:
using the `Identifier` class will remove a possible quoting (double quotes) of an identifier. 
`$this->quoteStringLiteral($table);` will quote the table name identifier for string literal usages(single quotes).

Reply:
alright :)